invenio_records_resources/services/base/service.py

```diff
@@ -60,11 +60,21 @@ def components(self):
 
     def run_components(self, action, *args, **kwargs):
         """Run components for a given action."""
-        # Run post commit components
         for component in self.components:
             if hasattr(component, action):
                 getattr(component, action)(*args, **kwargs)
 
+    def run_components_with_uow(self, action, uow, *args, **kwargs):
+        """Run components for a given action."""
+        for component in self.components:
+            if hasattr(component, action):
+                # Done like this to avoid API changes.
```

Comment:
Not pretty. Alternatively, we have to duplicate the ``components`` property.

Reply:
Do you refer to the line of code `component.uow = uow`? 
Instead of creating a new `run_components_with_uow`, can't you do in `run_components`:
```
uow  = kwargs.get("ouw")
if uow:
    component.uow = uow
    ....
```